### PR TITLE
test: moving to testing-library in `Tree`

### PR DESCRIPTION
**Pull request by dancerphil**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/tree/__tests__/directory.test.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import React from 'react';
-import { mount, render } from 'enzyme';
 import debounce from 'lodash/debounce';
-import Tree from '../index';
+import React from 'react';
```
Comment (dancerphil) [inline components/tree/__tests__/directory.test.js:2]: import 顺序看起来是自动的🤔

**Comment by zombieJ** [comment]
CI failed in React 18

**Comment by dancerphil** [comment]
rebased

**Comment by dancerphil** [comment]
@zombieJ

react 18 会把 css-motion 运行结束后再匹配 snapshot，所以 dom 结构中不含有 motion 的部分。

不确定为什么 17 和 18 会有这个区别，猜测和 `rc-motion` 有关（仅猜测）

我是否可以通过在测试中加入额外的 jest.runAllTimers() 跳过动画，使得 react 17 和 react 18 的测试保持一致。同时，这个应该更符合 test 的初衷（测最终结果而不是动画的中间状态）

**Pull request merged**